nsv13/code/modules/power/rbmk.dm

```diff
@@ -0,0 +1,864 @@
+			var/permeability_bonus = total_permeability_moles / 500
+			gas_absorption_effectiveness = gas_absorption_constant + permeability_bonus
+		var/total_degradation_moles = moderator_input.get_moles(/datum/gas/nitryl) //Because it's quite hard to get.
+		if(total_degradation_moles >= minimum_coolant_level*0.5) //I'll be nice.
+			depletion_modifier += total_degradation_moles / 15 //Oops! All depletion. This causes your fuel rods to get SPICY.
+			playsound(src, pick('sound/machines/sm/accent/normal/1.ogg','sound/machines/sm/accent/normal/2.ogg','sound/machines/sm/accent/normal/3.ogg','sound/machines/sm/accent/normal/4.ogg','sound/machines/sm/accent/normal/5.ogg'), 100, TRUE)
+		//From this point onwards, we clear out the remaining gasses.
+		moderator_input.clear() //Woosh. And the soul is gone.
+		K += total_fuel_moles / 1000
+	var/fuel_power = 0 //So that you can't magically generate K with your control rods.
+	if(!has_fuel())  //Reactor must be fuelled and ready to go before we can heat it up boys.
+		K = 0
+	else
+		for(var/obj/item/fuel_rod/FR in fuel_rods)
+			K += FR.fuel_power
+			fuel_power += FR.fuel_power
+			FR.deplete(depletion_modifier)
+	//Firstly, find the difference between the two numbers.
+	var/difference = abs(K - desired_k)
+	//Then, hit as much of that goal with our cooling per tick as we possibly can.
+	difference = CLAMP(difference, 0, control_rod_effectiveness) //And we can't instantly zap the K to what we want, so let's zap as much of it as we can manage....
+	if(difference > fuel_power && desired_k > K)
+		message_admins("Not enough fuel to get [difference]. We have fuel [fuel_power]")
+		difference = fuel_power //Again, to stop you being able to run off of 1 fuel rod.
+	if(K != desired_k)
+		if(desired_k > K)
+			K += difference
+		else if(desired_k < K)
+			K -= difference
+
+	K = CLAMP(K, 0, RBMK_MAX_CRITICALITY)
+	if(has_fuel())
+		temperature += K
+	else
+		temperature -= 10 //Nothing to heat us up, so.
+	handle_alerts() //Let's check if they're about to die, and let them know.
+	update_icon()
+	radiation_pulse(src, temperature*radioactivity_spice_multiplier)
+	if(power >= 90 && world.time >= next_flicker) //You're overloading the reactor. Give a more subtle warning that power is getting out of control.
+		next_flicker = world.time + 1.5 MINUTES
+		for(var/obj/machinery/light/L in GLOB.machines)
+			if(prob(25) && shares_overmap(src, L)) //If youre running the reactor cold though, no need to flicker the lights.
+				L.flicker()
+	for(var/atom/movable/I in get_turf(src))
+		if(isliving(I))
+			var/mob/living/L = I
+			if(temperature > 0)
+				L.adjust_bodytemperature(CLAMP(temperature, BODYTEMP_COOLING_MAX, BODYTEMP_HEATING_MAX)) //If you're on fire, you heat up!
+		if(istype(I, /obj/item/reagent_containers/food) && !istype(I, /obj/item/reagent_containers/food/drinks))
+			playsound(src, pick('sound/machines/fryer/deep_fryer_1.ogg', 'sound/machines/fryer/deep_fryer_2.ogg'), 100, TRUE)
```

Comment:
Can we have a spam checking here please to prevent it being too many items getting cooked at once? Producing so much sound/lag its stupid.
Also maybe it should mess with the reactor slightly somehow if you are doing a lot of stuff with it.